mvrx-rxjava3/src/main/kotlin/com/airbnb/mvrx/BaseMvRxViewModel.kt

```diff
@@ -0,0 +1,464 @@
+package com.airbnb.mvrx
+
+import android.annotation.SuppressLint
+import android.util.Log
+import androidx.annotation.RestrictTo
+import androidx.lifecycle.Lifecycle
+import androidx.lifecycle.LifecycleOwner
+import androidx.lifecycle.LifecycleRegistry
+import androidx.lifecycle.ViewModel
+import com.airbnb.mvrx.rxjava3.MvRxTuple1
+import com.airbnb.mvrx.rxjava3.MvRxTuple2
+import com.airbnb.mvrx.rxjava3.MvRxTuple3
+import com.airbnb.mvrx.rxjava3.MvRxTuple4
+import com.airbnb.mvrx.rxjava3.MvRxTuple5
+import com.airbnb.mvrx.rxjava3.MvRxTuple6
+import com.airbnb.mvrx.rxjava3.MvRxTuple7
+import io.reactivex.rxjava3.core.Completable
+import io.reactivex.rxjava3.core.Observable
+import io.reactivex.rxjava3.core.Single
+import io.reactivex.rxjava3.disposables.CompositeDisposable
+import io.reactivex.rxjava3.disposables.Disposable
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Job
+import kotlinx.coroutines.flow.distinctUntilChanged
+import kotlinx.coroutines.flow.launchIn
+import kotlinx.coroutines.flow.map
+import kotlinx.coroutines.flow.onEach
+import kotlin.reflect.KProperty1
+
+/**
+ * Base ViewModel implementation that all other ViewModels should extend.
+ */
+abstract class BaseMvRxViewModel<S : MavericksState>(
```

Comment:
some of this code doesn't need to be duplicated. each rx version can share a `InternalMvRxViewModel` super class which can be shared from a new module